pkg/helm/findscore.go

```diff
@@ -0,0 +1,162 @@
+		ChartName:   release.Chart.Metadata.Name,
+		Namespace:   release.Namespace,
+		Description: release.Chart.Metadata.Description,
+		Home:        release.Chart.Metadata.Home,
+		Icon:        release.Chart.Metadata.Icon,
+		Installed: output.VersionInfo{
+			Version:    release.Chart.Metadata.Version,
+			AppVersion: release.Chart.Metadata.AppVersion,
+		},
+		Latest: output.VersionInfo{
+			Version:    pkg.Version,
+			AppVersion: pkg.AppVersion,
+		},
+		IsOld:       version.Compare(release.Chart.Metadata.Version, pkg.Version, "<"),
+		Deprecated:  pkg.Deprecated,
+		HelmVersion: "3",
+	}
+}
+
+func scoreChartSimilarity(release *release.Release, pkg ArtifactHubHelmPackage) int {
+	ret := 0
+	var preferredRepositories = []string{
+		"bitnami",
+		"fairwinds-stable",
+		"ingress-nginx",
+	}
+	if release.Chart.Metadata.Home == pkg.HomeURL {
+		klog.V(10).Infof("+1 score for %s Home URL (ahub package repo %s)", release.Chart.Metadata.Name, pkg.Repository.Name)
+		ret++
+	}
+	if release.Chart.Metadata.Description == pkg.Description {
+		klog.V(10).Infof("+1 score for %s Description (ahub package repo %s)", release.Chart.Metadata.Name, pkg.Repository.Name)
+		ret++
+	}
+	for _, source := range pkg.Links {
+		if source.Name == "source" {
+			if containsString(release.Chart.Metadata.Sources, source.URL) {
+				klog.V(10).Infof("+1 score for %s source links (ahub package repo %s)", release.Chart.Metadata.Name, pkg.Repository.Name)
+				ret++
+			}
+		}
+	}
+	pkgMaintainers := map[string]bool{}
+	for _, m := range pkg.Maintainers {
+		pkgMaintainers[m.Email+";"+m.Name+";"] = true
+	}
+	matchedMaintainers := 0
+	for _, m := range release.Chart.Metadata.Maintainers {
+		if pkgMaintainers[m.Email+";"+m.Name+";"] {
+			matchedMaintainers++
+		}
+	}
+	if matchedMaintainers > 0 {
+		klog.V(10).Infof("+1 score for %s Maintainers (ahub package repo %s)", release.Chart.Metadata.Name, pkg.Repository.Name)
+		ret++
+	}
+	if pkg.Repository.VerifiedPublisher {
+		klog.V(10).Infof("+1 score for %s verified publisher (ahub package repo %s)", release.Chart.Metadata.Name, pkg.Repository.Name)
+		ret++
+	}
+	if clusterVersionExistsInPackage(release.Chart.Metadata.Version, pkg) {
+		klog.V(10).Infof("+1 score for %s, current version exists in available versions (ahub package %s)", release.Chart.Metadata.Name, pkg.Repository.Name)
+		ret++
+	}
+	if containsString(preferredRepositories, pkg.Repository.Name) {
+		klog.V(10).Infof("+1 score for %s, preffered repo (ahub package repo %s)", release.Chart.Metadata.Name, pkg.Repository.Name)
+		ret++
+	}
+	klog.V(10).Infof("calculated score repo: %s, release: %s, score: %d\n\n", pkg.Repository.Name, release.Name, ret)
+	return ret
+}
+
+func clusterVersionExistsInPackage(clusterVersion string, pkg ArtifactHubHelmPackage) bool {
+	for _, packageVersion := range pkg.AvailableVersions {
+		if packageVersion.Version == clusterVersion {
+			return true
+		}
+	}
+	return false
+}
+
+func containsString(arr []string, val string) bool {
```

Comment:
non-blocking: Highly recommend go-funk for this type of stuff. It's super useful.

Reply:
For the limited amount of usage here, I think I'll err on the side of not adding another dependency for now. If usage of this type of function expands in the future we can use go-funk.